src/main/java/bio/terra/service/snapshot/Snapshot.java

```diff
@@ -81,6 +81,11 @@ public Snapshot snapshotSources(List<SnapshotSource> snapshotSources) {
         return this;
     }
 
+    // TODO: When we support more than one dataset per snapshot, all uses of this will need refactoring.
+    public SnapshotSource getFirstSnapshotSource() {
```

Comment:
This is a totally minor thing, but to me the fact that this returns the `first` source is not part of the API, it's an implementation detail. As I understand it, the API is supposed to return _the_ snapshot source for a snapshot. So I would call this `getSnapshotSource()`. The caller is delegating the choice of which source to use to the `Snapshot` object.

Reply:
I'm using "first" here because we have a difference between the documentation and implementation. In the documentation, there's no single source for a snapshot: it could be one or many datasets. However, in our implementation, we have a 1:1 relationship between snaphot and dataset. "first" is descriptive of the behavior since its the first in the list of sources, which as of this point, is the only source. I agree that its weird, but I think "first" is the most descriptive.